tests/bluetooth/bsim/audio/src/tbs_client_test.c

```diff
@@ -255,14 +369,55 @@ static void test_main(void)
 	WAIT_FOR_COND(call_state == BT_TBS_CALL_STATE_ACTIVE);
 
 	printk("Reading bearer provider name\n");
+	UNSET_FLAG(provider_name);
 	err = bt_tbs_client_read_bearer_provider_name(default_conn, index);
 	if (err != 0) {
 		FAIL("Read bearer provider name failed (%d)\n", err);
 	}
 
 	test_ccid();
-
 	WAIT_FOR_COND(read_complete);
+
+	UNSET_FLAG(signal_strength);
+	printk("Reading signal strength\n");
+	err = bt_tbs_client_read_signal_strength(default_conn, index);
+	if (err != 0) {
+		FAIL("Read signal strength failed (%d)\n", err);
+	}
+
+	WAIT_FOR_FLAG(signal_strength);
+	printk("Signal strength read\n");
```

Comment:
Suggest to put these "blocks" into small functions like we've done with e.g. VCP. 

This make it easier to reuse them, as well as expanding the individual API tests with invalid behavior